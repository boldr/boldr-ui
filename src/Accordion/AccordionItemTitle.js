/* @flow */
import * as React from 'react';

type Props = {
  id: string,
  isExpanded: boolean,
  onClick: Function,
  ariaControls: string,
  children: Array<React.Node>,
  role: string,
};

class AccordionItemTitle extends React.PureComponent<Props, *> {
  static defaultProps = {
    id: '',
    isExpanded: false,
    onClick: () => {},
    ariaControls: '',
    role: '',
  };

  handleKeyPress = (e: Event) => {
    const { onClick } = this.props;
    if (e.charCode === 13 || e.charCode === 32) {
      onClick();
    }
  };
  props: Props;
  render() {
    const { id, isExpanded, ariaControls, onClick, children, role } = this.props;
    return (
      <div // eslint-disable-line jsx-a11y/no-static-element-interactions
        id={id}
        aria-expanded={isExpanded}
        aria-controls={ariaControls}
        className="boldrui-accordion__title"
        onClick={onClick}
        role={role}
        onKeyPress={this.handleKeyPress}
      >
        {children}
      </div>
    );
  }
}

export default AccordionItemTitle;
