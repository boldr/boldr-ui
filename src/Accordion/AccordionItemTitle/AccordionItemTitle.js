/* @flow */
import React, { PureComponent } from 'react';

type Props = {
  id: string,
  expanded: boolean,
  onClick: () => void,
  ariaControls: string,
  children: ReactChildren,
  role: string,
};

class AccordionItemTitle extends PureComponent {
  static defaultProps = {
    id: '',
    expanded: false,
    onClick: () => {},
    ariaControls: '',
    role: '',
  };

  handleKeyPress = (e: Event) => {
    const { onClick } = this.props;
    if (e.charCode === 13 || e.charCode === 32) {
      onClick();
    }
  };
  props: Props;
  render() {
    const { id, expanded, ariaControls, onClick, children, role } = this.props;
    return (
      <div // eslint-disable-line jsx-a11y/no-static-element-interactions
        id={id}
        aria-expanded={expanded}
        aria-controls={ariaControls}
        className="boldrui-accordion__title"
        onClick={onClick}
        role={role}
        onKeyPress={this.handleKeyPress}
      >
        {children}
      </div>
    );
  }
}
AccordionItemTitle.accordionElementName = 'AccordionItemTitle';

export default AccordionItemTitle;
