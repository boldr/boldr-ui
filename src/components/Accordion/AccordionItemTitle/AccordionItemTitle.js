/* @flow */
import React, { PureComponent } from 'react';
import type { ReactChildren } from '../../../types/react.js.flow';

type Props = {
  id: string,
  expanded: boolean,
  onClick: () => void,
  ariaControls: string,
  children: ReactChildren,
  className: string,
  role: string,
};

class AccordionItemTitle extends PureComponent {
  static defaultProps = {
    id: '',
    expanded: false,
    onClick: () => {},
    ariaControls: '',
    className: 'boldrui-accordion__title',
    role: '',
  };

  handleKeyPress = (e: Event) => {
    const { onClick } = this.props;
    if (e.charCode === 13 || e.charCode === 32) {
      onClick();
    }
  };
  props: Props;
  render() {
    const {
      id,
      expanded,
      ariaControls,
      onClick,
      children,
      className,
      role,
    } = this.props;
    return (
      <div // eslint-disable-line jsx-a11y/no-static-element-interactions
        id={id}
        aria-expanded={expanded}
        aria-controls={ariaControls}
        className={className}
        onClick={onClick}
        role={role}
        tabIndex="0"
        onKeyPress={this.handleKeyPress}
      >
        {children}
      </div>
    );
  }
}
AccordionItemTitle.accordionElementName = 'AccordionItemTitle';

export default AccordionItemTitle;
