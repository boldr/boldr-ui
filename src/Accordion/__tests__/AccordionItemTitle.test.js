import React from 'react';
import renderer from 'react-test-renderer';

import AccordionItemTitle from '../AccordionItemTitle';

describe('AccordionItemTitle', () => {
  it('renders correctly with min params', () => {
    const tree = renderer
      .create(
        <AccordionItemTitle>
          <div>Fake Title</div>
        </AccordionItemTitle>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with different className', () => {
    const tree = renderer
      .create(
        <AccordionItemTitle className="testCSSClass">
          <div>Fake Title</div>
        </AccordionItemTitle>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly when pressing enter', () => {
    const mockOnClick = jest.fn();
    const tree = renderer.create(
      <AccordionItemTitle onClick={mockOnClick}>
        <div>Fake Title</div>
      </AccordionItemTitle>,
    );

    tree.getInstance().handleKeyPress({ charCode: 13 });
    expect(mockOnClick).toHaveBeenCalledTimes(1);
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly when pressing space', () => {
    const mockOnClick = jest.fn();
    const tree = renderer.create(
      <AccordionItemTitle onClick={mockOnClick}>
        <div>Fake Title</div>
      </AccordionItemTitle>,
    );

    tree.getInstance().handleKeyPress({ charCode: 32 });
    expect(mockOnClick).toHaveBeenCalledTimes(1);
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly when pressing another key', () => {
    const mockOnClick = jest.fn();
    const tree = renderer.create(
      <AccordionItemTitle onClick={mockOnClick}>
        <div>Fake Title</div>
      </AccordionItemTitle>,
    );

    tree.getInstance().handleKeyPress({ charCode: 35 });
    expect(mockOnClick).toHaveBeenCalledTimes(0);
    expect(tree).toMatchSnapshot();
  });
});
