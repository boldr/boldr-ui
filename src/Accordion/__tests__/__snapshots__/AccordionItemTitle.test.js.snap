// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccordionItemTitle renders correctly when pressing another key 1`] = `
<div
  aria-controls=""
  aria-expanded={false}
  className="boldrui-accordion__title"
  id=""
  onClick={[Function]}
  onKeyPress={[Function]}
  role=""
>
  <div>
    Fake Title
  </div>
</div>
`;

exports[`AccordionItemTitle renders correctly when pressing enter 1`] = `
<div
  aria-controls=""
  aria-expanded={false}
  className="boldrui-accordion__title"
  id=""
  onClick={[Function]}
  onKeyPress={[Function]}
  role=""
>
  <div>
    Fake Title
  </div>
</div>
`;

exports[`AccordionItemTitle renders correctly when pressing space 1`] = `
<div
  aria-controls=""
  aria-expanded={false}
  className="boldrui-accordion__title"
  id=""
  onClick={[Function]}
  onKeyPress={[Function]}
  role=""
>
  <div>
    Fake Title
  </div>
</div>
`;

exports[`AccordionItemTitle renders correctly with different className 1`] = `
<div
  aria-controls=""
  aria-expanded={false}
  className="boldrui-accordion__title"
  id=""
  onClick={[Function]}
  onKeyPress={[Function]}
  role=""
>
  <div>
    Fake Title
  </div>
</div>
`;

exports[`AccordionItemTitle renders correctly with min params 1`] = `
<div
  aria-controls=""
  aria-expanded={false}
  className="boldrui-accordion__title"
  id=""
  onClick={[Function]}
  onKeyPress={[Function]}
  role=""
>
  <div>
    Fake Title
  </div>
</div>
`;
